frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx

```diff
@@ -51,23 +51,29 @@ function PinToPlaylistButton({
         description = 'Save'
     }
 
-    return logicProps.setPinned ? (
+    return logicProps.setPinned && !logicProps.pinned ? (
```

Comment:
What I went with is:

- If we're not not in a playlist view, keep everything as-is
- If we're in a playlist, and the recording is not pinned, keep everything as is (I'd imagine the default action for most would be to want to pin to the current playlist)
- **If we're in a playlist, and the recording is pinned, open the popover instead.**

Reply:
that's works nicely...

I think we need to dig into how people are using playlists and whether we can break the "saved filters" vs "pinned recordings" overlap

but don't need to do that here!